.github/workflows/main.yml

```diff
@@ -29,5 +29,33 @@ jobs:
       - name: Espresso Checks
         uses: reactivecircus/android-emulator-runner@v2.13.0
         with:
-          api-level: 23
+          api-level: 16
           script: ./gradlew leakcanary-android-core:connectedCheck leakcanary-android-instrumentation:connectedCheck --stacktrace
+
+  snapshot-deployment:
```

Comment:
I compared with https://github.com/square/okhttp/blob/master/.github/workflows/build.yml  , let me know what you think:

* Should we really trigger snapshots deploy on all pushes? Seems like we should target master only, e.g.`if: github.ref == 'refs/heads/master'` 
* `runs-on: ubuntu-20.04` vs `runs-on: ubuntu-latest` ? Also, why are we using `runs-on: macOS-latest` for the `checks` job?
* what's the difference between `actions/checkout@v2.3.4` and `actions/checkout@v2` ? Does `@v2` mean "latest minor in v2", in which case should we do that?
* Why write SONATYPE_NEXUS_USERNAME & SONATYPE_NEXUS_PASSWORD to `~/.gradle/gradle.properties` ? `env` should be enough, no?
* `./gradlew clean uploadArchives` => `./gradlew uploadArchives`  (this should already be clean)

Reply:
- The workflow itself is gated on pushes to the main branch, so it won't execute on every push and checking the ref is redundant.

- GitHub is moving the `ubuntu-latest` identifier to Ubuntu 20.04 in an incremental rollout, and drops a warning about it in your Actions logs if you use `ubuntu-latest`. My use of 20.04 directly was done simply to avoid the warning and directly move to the new base.

- The GHA ecosystem has settled on a versioning system where v2 means the latest minor release, so v2 will be a tag that gets force pushed every time a new minor/patch release happens. GitHub has respected semver in my experience so we can probably switch to v2 without the workflow breaking when GitHub changes the tag next.

- I took a quick glance at the gradle-mvn-push.gradle file and didn't see an environment fallback, and the releasing docs said to do this instead so I simply replicated that.

- Yep, will drop the extraneous clean.

Reply:
I've switched the workflow to `ubuntu-latest`, changed all actions to point to their `v$major` tags, replaced the filling of properties into `$HOME/gradle/gradle.properties` with the appropriate Gradle-ism for loading properties from the environment, and dropped the unnecessary `clean` task invocation. That should address everything I believe.

Reply:
Thx! "GitHub drops a warning about it in your Actions logs if you use ubuntu-latest" vs "I've switched the workflow to ubuntu-latest" => I'm curious, why did you change it then? Sounds like you were saying it's best if we don't?

I honestly don't have a ton of context so I was mostly asking questions :)

Reply:
I switched it because I noticed that the workflow runs perfectly fine on both Ubuntu 18.04 and 20.04 so it wouldn't affect us when the migration does happen, and it's just less maintenance work over a longer period when the next LTS of Ubuntu becomes available.